### PR TITLE
Add `ignore: ["with-var-inside"]` to `color-function-notation`

### LINKED ISSUE #6620: Add `ignore: ["with-var-inside"]` to `color-function-notation`
### What steps are needed to reproduce the bug?

```css
div {
  background-color: rgba(var(--bs-primary-rgb), 0.4);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "ignoreFiles": ["**/*"],
  "overrides": [
    {
      "files": ["**/*.css"],
      "extends": ["stylelint-config-standard", "stylelint-config-prettier"],
      "rules": {}
    },
    {
      "files": ["**/*.scss"],
      "extends": ["stylelint-config-standard-scss", "stylelint-config-prettier"],
      "rules": {}
    }
  ],
  "rules": {}
}
```

### How did you run Stylelint?

`stylelint app/styles`

### Which version of Stylelint are you using?

14.16.1

### What did you expect to happen?

No error/warning showing since the syntax is correct.

### What actually happened?

An error is reported.

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

Maybe related to https://github.com/stylelint/stylelint/issues/5671.

### PR DESCRIPTION
**Pull request by mattxwang**
> Which issue, if any, is this issue related to?

Closes #6620.

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/color-function-notation/README.md [hunk 1]
```diff
@@ -116,3 +116,37 @@ a { color: hsla(270, 60%, 50%, 15%) }
...
 a { color: hsl(.75turn, 60%, 70%) }
 ```
+
+## Optional secondary options
+
+### `ignore: ["with-var-inside"]`
+
+Ignore color functions containing variables.
+
+Given
```
Comment (ybiquitous) [inline lib/rules/color-function-notation/README.md:126]: Typo.

```suggestion
Given:
```

File: lib/rules/color-function-notation/README.md [hunk 2]
```diff
@@ -116,3 +116,37 @@ a { color: hsla(270, 60%, 50%, 15%) }
...
+
+The following patterns are _not_ considered problems:
+
+```css
+a {
+  color: rgba(var(--foo), 0.5);
+}
+```
+
+Given
```
Comment (ybiquitous) [inline lib/rules/color-function-notation/README.md:140]: Typo.

```suggestion
Given:
```

File: lib/rules/color-function-notation/index.js [hunk 3]
```diff
@@ -46,6 +60,13 @@ const rule = (primary, _secondaryOptions, context) => {
 
 				const { value, sourceIndex, sourceEndIndex, nodes } = node;
 
+				if (
+					ignoreWithVarInside &&
+					nodes.filter((child) => child.type === 'function' && child.value.toLowerCase() === 'var')
+						.length > 0
+				)
+					return;
```
Comment (ybiquitous) [inline lib/rules/color-function-notation/index.js:68]: [suggest] I recommend using `.some()` over `.filter().length`. Also, using `{}` for `return` at a new line seems more readable.

```suggestion
				if (
					ignoreWithVarInside &&
					nodes.some((child) => child.type === 'function' && child.value.toLowerCase() === 'var')
				) {
					return;
				}
```

Comment (ybiquitous) [reply lib/rules/color-function-notation/index.js:68]: Or, extracting a helper function might be more readable like this:

```js
if (ignoreWithVarInside && containsVariable(nodes)) return;

// ...

/**
 * @param {import('postcss-value-parser').Node[]} nodes
 */
function containsVariable(nodes) {
	return nodes.some(({ type, value }) => type === 'function' && value.toLowerCase() === 'var');
}
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@mattxwang Thanks for the pull request! I've commented on refactoring, but this option addition is nice! 👍🏼

**Comment by mattxwang** [comment]
Thanks for the feedback! I think I've addressed all the code suggestions.

**Review by ybiquitous — APPROVED** [review]
Thanks, LGTM 👍🏼

**Comment by mattxwang** [comment]
(dup comment as #6778)

> Shall I hold off on merging this since #6795 is a patch?

**Comment by ybiquitous** [comment]
Let's merge it. See https://github.com/stylelint/stylelint/issues/6795#issuecomment-1517018386

**Pull request merged**